### PR TITLE
STORM_B (feat): add user profile contoller

### PR DESCRIPTION
**Pull request by Shabaldas**
This pull request introduces a new feature to the application, allowing users to view and edit their profiles. The changes include the creation of a new `ProfilesController`, profile views, and the addition of profile-related routes. The most significant changes are:

New Controller:

* [`app/controllers/profiles_controller.rb`](diffhunk://#diff-7a47d335107fc35c2e79cc9c2b0a195c863594dd2343bcd48a4c770ac4af673cR1-R26): A new `ProfilesController` has been added to handle profile-related requests. This controller includes `show`, `edit`, and `update` actions, which find the user based on the `id` parameter and perform the corresponding actions.

New Views:

* [`app/views/profiles/_form.html.erb`](diffhunk://#diff-55d9b525aa656234d5e16ddf9cebe33ef647e9e0d996edbe071a0d86d71f9e30R1-R34): A new form partial view has been added to display and edit user profile information, including the first and last names.
* [`app/views/profiles/edit.html.erb`](diffhunk://#diff-6091542e4570d68ca546a4a893b0d0202a75b9aa6fabf4e8aa0b4122ae6a3544R1-R28): An edit view has been added that uses the form partial to allow users to edit their profile information.
* [`app/views/profiles/show.html.erb`](diffhunk://#diff-64807e65095290f37cffcfa9232f887f63a6cd7d3d36c764c3bb04e4bd65b04eR1-R15): A show view has been added to display the user's profile information.

Routes and Navigation:

* [`config/routes.rb`](diffhunk://#diff-959bc9abc46a55332bb64d5155a79323afa75a50ec1a2137ddd22d926f62c6c5R40-R41): New routes have been added to handle `show`, `edit`, and `update` requests for profiles.
* [`app/views/shared/_header.html.erb`](diffhunk://#diff-800d7323f71109e9501d9a597745a786062dc45970b87f34caf50e06a19e904dR55-R57): A new link to the profile page has been added to the header for signed-in users.

image
image

### CODE REVIEW

File: spec/request/profile_spec.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    let(:params_invalid) do
+      {
+        user: {
+          password: '1111112',
+          password_confirmation: '111111'
+        }
+      }
+    end
+
+    it 'update user personal data' do
```
Comment (Shabaldas) [inline spec/request/profile_spec.rb:59]: wronf naming

Comment (AndriiBuhalo) [reply spec/request/profile_spec.rb:59]: +

Comment (Shabaldas) [reply spec/request/profile_spec.rb:59]: > done

where?

Comment (AndriiBuhalo) [reply spec/request/profile_spec.rb]: Look in my code

**Pull request merged**